listenbrainz/spotify_updater/spotify_read_listens.py

```diff
@@ -125,26 +125,28 @@ def _convert_spotify_play_to_listen(play, listen_type):
     return listen
 
 
-def make_api_request(user, spotipy_call: callable, **kwargs):
+def make_api_request(user: dict, endpoint: str, **kwargs):
     """ Make an request to the Spotify API for particular user at specified endpoint with args.
 
     Args:
-        user (spotify.Spotify): the user whose plays are to be imported.
-        spotipy_call (callable): the Spotipy function which makes request to the required API endpoint
+        user: the user whose plays are to be imported.
+        endpoint: the name of Spotipy function which makes request to the required API endpoint
 
     Returns:
         the response from the spotify API
 
     Raises:
-        spotify.SpotifyAPIError: if we encounter errors from the Spotify API.
-        spotify.SpotifyListenBrainzError: if we encounter a rate limit, even after retrying.
+        ExternalServiceAPIError: if we encounter errors from the Spotify API.
+        ExternalServiceError: if we encounter a rate limit, even after retrying.
     """
     retries = 10
     delay = 1
     tried_to_refresh_token = False
 
     while retries > 0:
         try:
+            spotipy_client = spotipy.Spotify(auth=user['access_token'])
+            spotipy_call = getattr(spotipy_client, endpoint)
             recently_played = spotipy_call(**kwargs)
```

Comment:
these could fail if the endpoint doesn't exist on the client, or if the kwargs aren't valid for that call. We should catch these specific cases and log them internally (e.g. imagine if spotipy changes the api at some point)